mono/metadata/icall.c

```diff
@@ -6205,6 +6205,12 @@ ves_icall_System_Delegate_CreateDelegate_internal (MonoReflectionType *type, Mon
 		}
 	}
 
+	if (method->klass->class_kind == MONO_CLASS_GTD)
+	{
+		mono_set_pending_exception(mono_get_exception_argument("method", " Cannot bind to the target method because the type is not inflated"));
```

Comment:
Is this close to exception from .NET?

Reply:
It might be nice to name the type as well in the exception message.